pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -333,6 +288,51 @@ abstract class Analytics {
   ///
   /// The snooze period is defined by the [Survey.snoozeForMinutes] field.
   void surveyShown(Survey survey);
+
+  /// Returns an instance of [FakeAnalytics] which can be used in tests to check
+  /// for certain [Event] instances within [FakeAnalytics.sentEvents].
+  @visibleForTesting
+  static FakeAnalytics fake({
```

Comment:
can you add some context on why we're changing it from a factory to a static? I don't understand the motivation for this change.

Reply:
This is so that we don't have to cast to `FakeAnalytics` when we use the `Analytics.test` constructor.

It was pointed out in Gerrit CL comment here: https://dart-review.googlesource.com/c/sdk/+/360540/comment/58e6bf9a_739d6c51/